icechunk/src/format/manifest.rs

```diff
@@ -186,28 +251,42 @@ impl Manifest {
 
         let mut all = all.iter().peekable();
 
-        let mut array_manifests = Vec::with_capacity(1);
+        let mut array_manifests = Vec::with_capacity(shards.len());
+        // why peek?
+        // this seems to handle multiple nodes, but we only send one in?
```

Comment:
because we need to gather chunks for each node (this handles the case of chunks coming from multiple arrays to the same manifest). So peek allows me to know if I should continue or if I can finalize the current node.